### PR TITLE
initial

### PR DESCRIPTION
**Pull request by zyndiecate**
Library for generic type processing. Currently supported types are:

- Domain
- DockerPort
- DockerImage

### CODE REVIEW

**Comment by zyndiecate** [comment]
ping @ZeissS @hectorj2f @denderello

File: README.md [hunk 1]
```diff
@@ -1 +1,3 @@
 # generic-types-go
+
+Generic types processing in Go
```
Comment (ZeissS) [inline README.md]: Can we add another paragraph with the idea?

> This repository is intended to house generally usable types that are missing from the standard library (Domains) or a bit more specific for the containerized world we live in (DockerImage, DockerPort).
> All types should support JSON serialization and a validation logic.

File: domain.go [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+	"github.com/alexanderritola/validate"
+	"github.com/alexanderritola/validate/web"
+	"github.com/giantswarm/generic-types-go"
+	"github.com/juju/errgo"
+)
+
+type Domain string
+
+type DomainConfig map[Domain]generictypes.DockerPort
```
Comment (ZeissS) [inline domain.go]: `DomainConfig` is very `user-config` library specific. I would leave that over there.

File: domain_test.go [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+import (
+	"encoding/json"
+	"testing"
+
+	"github.com/giantswarm/generic-types-go"
+	"github.com/giantswarm/user-config"
+)
+
+func TestUnmarshalValidDomains(t *testing.T) {
```
Comment (ZeissS) [inline domain_test.go]: All of these tests should not be in here. Especially not import our private repos ;)

Comment (zyndiecate) [inline domain_test.go]: Removed.

File: domain.go (hunk 2 above)
Comment (zyndiecate) [inline domain.go]: Right. Removed.

File: README.md (hunk 1 above)
Comment (zyndiecate) [inline README.md]: Copied that.

**Comment by ZeissS** [comment]
LGTM

**Pull request merged**